pkg/ottl/ottlfuncs/func_delete_key.go

```diff
@@ -17,23 +17,35 @@ package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-c
 import (
 	"context"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
-func DeleteKey[K any](target ottl.Getter[K], key string) (ottl.ExprFunc[K], error) {
+type DeleteKeyArguments[K any] struct {
+	Target ottl.Getter[K] `arg:"0"`
+	Key    string         `arg:"1"`
+}
+
+type DeleteKeyFactory[K any] struct{}
+
+func (f DeleteKeyFactory[K]) Name() string { return "delete_key" }
+
+func (f DeleteKeyFactory[K]) CreateEmptyArguments() ottl.Arguments {
+	return &DeleteKeyArguments[K]{}
+}
+
+func (f DeleteKeyFactory[K]) CreateFunction(ctx context.Context, set component.TelemetrySettings, args ottl.Arguments) (ottl.ExprFunc[K], error) {
+	dkargs := ottl.ConvertArguments[DeleteKeyArguments[K]](args)
```

Comment:
Is this function necessary?  As a function creator we know that that our `CreateEmptyArguments` returned a pointer we now that `args` here will also be a pointer right?

Reply:
The intent of using it in functions defined in the OTTL package would be to establish a clear pattern for when to use it, even if we already know we're returning a pointer value for the arguments.

I've pushed a commit that makes it a hard requirement to return a pointer value from `CreateEmptyArguments`, let me know what you think. As you've previously mentioned, returning struct values isn't caught by the compiler, but I think it would be reasonable for us to simply state that this is a requirement and gracefully handle it when the requirement isn't followed.